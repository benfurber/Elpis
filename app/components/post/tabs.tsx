import React from "react";
import { StyleSheet, TouchableOpacity, View } from "react-native";

import { colours, layout } from "styles";
import { Icon } from "components";

const iconSize = 30;

interface Props {
  onPressComments: () => any;
  onPressPost: () => any;
}

const Tabs = (props: Props) => {
  return (
    <View style={styles.tabs}>
      <View style={[styles.tabWithBackground, styles.tabSelected]}>
        <TouchableOpacity onPress={props.onPressPost}>
          <Icon name="palette" size={iconSize} style={styles.iconCentre} />
        </TouchableOpacity>
      </View>
      <View style={styles.tabWithBackground}>
        <TouchableOpacity onPress={props.onPressComments}>
          <Icon
            colour={colours.navyBlueDarkTransparentHigh}
            name="comments"
            size={iconSize}
            style={styles.iconCentre}
          />
        </TouchableOpacity>
      </View>
      <View style={styles.tabWithoutBackground}>
        <TouchableOpacity>
          <Icon
            name="ellipsis-h"
            size={iconSize - 5}
            style={styles.iconRight}
          />
        </TouchableOpacity>
      </View>
    </View>
  );
};

const tabs = {
  borderTopLeftRadius: layout.borderRadius,
  borderTopRightRadius: layout.borderRadius,
  flex: 1,
  marginRight: 2,
  paddingHorizontal: 16,
  paddingVertical: 8
};

const styles = StyleSheet.create({
  iconCentre: {
    alignSelf: "center"
  },
  iconRight: {
    alignSelf: "flex-end"
  },
  tabs: {
    alignContent: "center",
    alignItems: "center",
    flexDirection: "row",
    width: "100%"
  },
  tabSelected: {
    backgroundColor: colours.whiteTransparent
  },
  tabWithBackground: {
    ...tabs,
    backgroundColor: colours.whiteTransparentHigh
  },
  tabWithoutBackground: {
    ...tabs
  }
});

export { Tabs };
