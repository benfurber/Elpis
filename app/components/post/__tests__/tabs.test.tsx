import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { Tabs } from "../tabs";

describe("Tabs", () => {
  it("renders correctly", () => {
    const onPressComments = jest.fn();
    const onPressPost = jest.fn();

    const component = shallow(
      <Tabs onPressComments={onPressComments} onPressPost={onPressPost} />
    );

    expect(component).toMatchSnapshot();
  });
});
