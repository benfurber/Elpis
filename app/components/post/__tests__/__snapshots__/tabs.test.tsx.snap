// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    onPressComments={[MockFunction]}
    onPressPost={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.60)",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flex": 1,
                "marginRight": 2,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.85)",
              },
            ]
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
          >
            <customIcon
              name="palette"
              size={30}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            />
          </TouchableOpacity>
        </View>,
        <View
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.60)",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "flex": 1,
              "marginRight": 2,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
          >
            <customIcon
              colour="rgba(45, 82, 113, 0.60)"
              name="comments"
              size={30}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            />
          </TouchableOpacity>
        </View>,
        <View
          style={
            Object {
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "flex": 1,
              "marginRight": 2,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <customIcon
              name="ellipsis-h"
              size={25}
              style={
                Object {
                  "alignSelf": "flex-end",
                }
              }
            />
          </TouchableOpacity>
        </View>,
      ],
      "style": Object {
        "alignContent": "center",
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
          >
            <customIcon
              name="palette"
              size={30}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            />
          </TouchableOpacity>,
          "style": Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.60)",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "flex": 1,
              "marginRight": 2,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            },
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.85)",
            },
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <customIcon
              name="palette"
              size={30}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            />,
            "onPress": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "palette",
              "size": 30,
              "style": Object {
                "alignSelf": "center",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
            onPress={[MockFunction]}
          >
            <customIcon
              colour="rgba(45, 82, 113, 0.60)"
              name="comments"
              size={30}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            />
          </TouchableOpacity>,
          "style": Object {
            "backgroundColor": "rgba(255, 255, 255, 0.60)",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flex": 1,
            "marginRight": 2,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <customIcon
              colour="rgba(45, 82, 113, 0.60)"
              name="comments"
              size={30}
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            />,
            "onPress": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "colour": "rgba(45, 82, 113, 0.60)",
              "name": "comments",
              "size": 30,
              "style": Object {
                "alignSelf": "center",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.2}
          >
            <customIcon
              name="ellipsis-h"
              size={25}
              style={
                Object {
                  "alignSelf": "flex-end",
                }
              }
            />
          </TouchableOpacity>,
          "style": Object {
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flex": 1,
            "marginRight": 2,
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": <customIcon
              name="ellipsis-h"
              size={25}
              style={
                Object {
                  "alignSelf": "flex-end",
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "name": "ellipsis-h",
              "size": 25,
              "style": Object {
                "alignSelf": "flex-end",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0.60)",
                  "borderTopLeftRadius": 5,
                  "borderTopRightRadius": 5,
                  "flex": 1,
                  "marginRight": 2,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                },
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0.85)",
                },
              ]
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[MockFunction]}
            >
              <customIcon
                name="palette"
                size={30}
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              />
            </TouchableOpacity>
          </View>,
          <View
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.60)",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flex": 1,
                "marginRight": 2,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[MockFunction]}
            >
              <customIcon
                colour="rgba(45, 82, 113, 0.60)"
                name="comments"
                size={30}
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              />
            </TouchableOpacity>
          </View>,
          <View
            style={
              Object {
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flex": 1,
                "marginRight": 2,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
            >
              <customIcon
                name="ellipsis-h"
                size={25}
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              />
            </TouchableOpacity>
          </View>,
        ],
        "style": Object {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[MockFunction]}
            >
              <customIcon
                name="palette"
                size={30}
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              />
            </TouchableOpacity>,
            "style": Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.60)",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flex": 1,
                "marginRight": 2,
                "paddingHorizontal": 16,
                "paddingVertical": 8,
              },
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0.85)",
              },
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <customIcon
                name="palette"
                size={30}
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              />,
              "onPress": [MockFunction],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "palette",
                "size": 30,
                "style": Object {
                  "alignSelf": "center",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
              onPress={[MockFunction]}
            >
              <customIcon
                colour="rgba(45, 82, 113, 0.60)"
                name="comments"
                size={30}
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              />
            </TouchableOpacity>,
            "style": Object {
              "backgroundColor": "rgba(255, 255, 255, 0.60)",
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "flex": 1,
              "marginRight": 2,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <customIcon
                colour="rgba(45, 82, 113, 0.60)"
                name="comments"
                size={30}
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              />,
              "onPress": [MockFunction],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "colour": "rgba(45, 82, 113, 0.60)",
                "name": "comments",
                "size": 30,
                "style": Object {
                  "alignSelf": "center",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.2}
            >
              <customIcon
                name="ellipsis-h"
                size={25}
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              />
            </TouchableOpacity>,
            "style": Object {
              "borderTopLeftRadius": 5,
              "borderTopRightRadius": 5,
              "flex": 1,
              "marginRight": 2,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <customIcon
                name="ellipsis-h"
                size={25}
                style={
                  Object {
                    "alignSelf": "flex-end",
                  }
                }
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "name": "ellipsis-h",
                "size": 25,
                "style": Object {
                  "alignSelf": "flex-end",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
